odoo/models.py

```diff
@@ -3212,6 +3212,9 @@ def _check_company(self, fnames=None):
         inconsistencies = []
         for record in self:
             company = record.company_id if record._name != 'res.company' else record
+            # If the company is False, we should skip the check
+            if not company:
+                continue
```

Comment:
AFAIK, if a record doesn't belong to a specific company, we do not want to accept company-specific records to be linked to it, unless it is a `company_dependent` field.

The current codebase may not respect this logic strictly, but we shouldn't modify the core mechanism because of it.
For your specific case, I believe a specific exception (Cf. #64566 ) should be more appropriate.

cc: @rco-odoo

Reply:
I'm not sure if I put the verification in the right place, what do you think ?